Chapter 13 Outline (#102)
merely-useful/r-rse: chapters/13-package-release.Rmd

- Managing your versions, git tagging, DOI?
- ORCID here?

## Learning Objectives

> - Describe the importance of using a build system for validating softwate

Reviewer: ```suggestion
- Describe the importance of using a build system for validating software and how to apply it
```

> - Identify ethical guidelines when using build systems like RHub

Reviewer: ```suggestion
- Understand and implement ethical guidelines when using build systems like RHub
```

- Prepare a package version release on GitHub
- Manage current package versions and devise future ones

## Outline

* RHub
    * Text: What is `rhub`?
    * Text: Talk about why it is important to check in multiple environments
        * Text: Cover dependencies, file management, build processes
    * Exercise: Create an account in RHub and validate your email
        * `rhub::validate_email("youremail@email.com")`
    * Text: Talk about behaviour when using RHub and other build systems
        Exercise: Read https://docs.r-hub.io/ and answer "Can you use RHub to continously build and deploy your package?"

* GitHub Release Process
    * Text: Talk about Git tagging

>     * Text: Talk about GitHub releases

Reviewer: ```suggestion
    * Text: Talk about GitHub releases
    * Code: use_github_release(), use_release_issue()
```

    * Exercise: Perform a GitHub release of the core package

* Version management

>     * Text: When should I upgrade the package version?

Reviewer: ```suggestion
    * Text: When should I upgrade the package version?
    * Code: use_version(), use_dev_version()
```
